Switch C# codegen to schema (#1132)
pulumi/pulumi-kubernetes: CHANGELOG.md

## HEAD (Unreleased)

### Improvements

> - .NET SDK refactored to align with other .NET SDKs, e.g. ones based on Terraform. (https://github.com/pulumi/pulumi-kubernetes/pull/1132)

Reviewer: ```suggestion
- .NET SDK updated to align with other Pulumi .NET SDKs. (https://github.com/pulumi/pulumi-kubernetes/pull/1132)
```

>     - Older resources are marked as obsolete in accordance to SDKs for other languages.

Reviewer: ```suggestion
    - Deprecated resources are now marked as `Obsolete`.
```

>     - Many classes are moved to new locations while preserving the namespaces and API.

Reviewer: ```suggestion
    - Many classes are moved to new locations on disk while preserving the public namespaces and API.
```
